src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs

```diff
@@ -1094,13 +1114,24 @@ private ConsoleKeyInfo ReadKey(bool intercept)
                     // we can subscribe in the same way.
                     DebugServer?.SendNotification("powerShell/sendKeyPress");
                 });
-
+        
             // PSReadLine doesn't tell us when CtrlC was sent. So instead we keep track of the last
             // key here. This isn't functionally required, but helps us determine when the prompt
             // needs a newline added
             //
             // TODO: We may want to allow users of PSES to override this method call.
-            _lastKey = System.Console.ReadKey(intercept);
+            if (!_hostInfo.UseHostReadKey)
+            {
+                _lastKey = System.Console.ReadKey(intercept);
+            }
+            else
+            {
+                KeyInfo keyInfo = _hostInfo.PSHost.UI.RawUI.ReadKey();
```

Comment:
Silly question: Could we just make the default `RawUI.ReadKey` in PSES be `Console.ReadKey()`? I ask because I see that's how you're supplying a custom `ReadKey`, which is exactly what we're also trying to do.